izumi-reflect/izumi-reflect/src/main/scala/izumi/reflect/macrortti/LightTypeTagRef.scala

```diff
@@ -107,10 +107,11 @@ object LightTypeTagRef extends LTTOrdering {
       }
     }
 
-    private[this] def makeFakeParams: List[(LambdaParamName, NameReference)] = {
+    private[this] def makeNormalizedParams: List[(LambdaParamName, NameReference)] = {
       input.zipWithIndex.map {
         case (p, idx) =>
-          p -> NameReference(SymName.LambdaParamName(idx, -2, input.size)) // s"!FAKE_$idx"
+          val relativeDepth = -1 * (input.size - idx)
```

Comment:
Relative depth should be based on the nestedness of lambdas, not arity or current parameter index